sql/expression_resolver.go

```diff
@@ -128,152 +141,114 @@ type embeddingColumn struct {
 	Combiner       string
 }
 
-func resolveTrainClause(tc *trainClause) (*resolvedTrainClause, error) {
-	modelName := tc.estimator
-	preMadeModel := !strings.ContainsAny(modelName, ".")
-
-	// default value
-	batchSize := 512
-	dropRemainder := true
-	enableCache := false
-	cachePath := ""
-	epoch := 1
-	shard := 1
-	enableShuffle := false
-	shuffleBufferSize := 10240
-	maxSteps := -1
-	evalSteps := -1
-	evalStartDecaySecs := 120
-	evalThrottleSecs := 600
-	// end default
-
-	attrs, err := resolveTrainAttribute(&tc.attrs)
-	if err != nil {
-		return nil, err
-	}
-	modelParams := filter(attrs, "model", true)
-	trainParams := filter(attrs, "train", false)
-	evalParams := filter(attrs, "eval", false)
-
-	if p, ok := trainParams["batch_size"]; ok {
-		strVal, _ := p.Value.(string)
-		intVal, err := strconv.Atoi(strVal)
-		if err == nil {
-			batchSize = intVal
-		} else {
-			fmt.Printf("ignore invalid train.batch_size=%s, default is %d", p.Value, batchSize)
-		}
-		delete(attrs, p.FullName)
-	}
-
-	if p, ok := trainParams["drop_remainder"]; ok {
-		strVal, _ := p.Value.(string)
-		boolVal, err := strconv.ParseBool(strVal)
-		if err == nil {
-			dropRemainder = boolVal
-		} else {
-			fmt.Printf("ignore invalid train.drop_remainder=%s, default is %v", p.Value, dropRemainder)
-		}
-		delete(attrs, p.FullName)
-	}
+func getEngineSpec(attrs map[string]*attribute) engineSpec {
+	getInt := func(key string, defaultValue int) int {
+		if p, ok := attrs[key]; ok {
+			strVal, _ := p.Value.(string)
+			intVal, err := strconv.Atoi(strVal)
 
-	if p, ok := trainParams["cache"]; ok {
-		strVal, _ := p.Value.(string)
-		boolVal, err := strconv.ParseBool(strVal)
-		if err == nil {
-			enableCache = boolVal
-		} else {
-			//TODO(uuleon) validation for cachePath
-			enableCache = true
-			cachePath = strVal
+			if err == nil {
+				return intVal
+			}
 		}
-		delete(attrs, p.FullName)
+		return defaultValue
 	}
-
-	if p, ok := trainParams["epoch"]; ok {
-		strVal, _ := p.Value.(string)
-		intVal, err := strconv.Atoi(strVal)
-		if err == nil {
-			epoch = intVal
-		} else {
-			fmt.Printf("ignore invalid train.epoch=%s, default is %d", p.Value, epoch)
+	psNum := getInt("ps_num", 1)
+	psMemory := getInt("ps_memory", 2400)
+	workerMemory := getInt("worker_memory", 1600)
+	workerNum := getInt("worker_num", 2)
+	engineType := "local"
+	if p, ok := attrs["type"]; ok {
+		strVal, ok := p.Value.(string)
+		if ok {
+			engineType = strVal
 		}
-		delete(attrs, p.FullName)
+	} else if psNum > 0 || workerNum > 0 {
+		engineType = "k8s"
 	}
+	return engineSpec{
```

Comment:
Is there a way to let the submitter just return the yarn job URL and exit? then the status can be fetched on the web page and do not wait on the sqlflow client (e.g. jupyter notebook)

Reply:
I can modify to exit immediately, but how the web page can get the status?